src/editions/transitioning.md

```diff
@@ -1,11 +1,74 @@
 # Transitioning your code to a new edition
 
-In general, to update your code to a new edition, the process
-will look like this:
+Transitioning between editions is built around lints. Fundamentally, the
+process works like this:
 
-* Make sure your build is warning-free.
-* Update or add the `edition` key in `Cargo.toml`
-* Fix any warnings, possibly by running `rustfix`.
+* Get your code compiling with no warnings.
+* Opt in to the new edition.
+* Fix any new warnings that may result.
 
-At this time, the `edition` key does not yet work, and `rustfix` is in a
-larval state. This section will update as our story gets better here.
\ No newline at end of file
+Luckily, we've been working on a tool to help assist with this process,
+`rustfix`. It can take suggestions from the compiler and automatically
+re-write your code to comply with new features and idioms.
+
+> `rustfix` is still quite young, and very much a work in development. But it works
+> for the basics! We're working hard on making it better and more robust, but
+> please bear with us for now.
+
+## Installing rustfix
+
+You can get `rustfix` from crates.io. Given that you're probably using Cargo,
+you'll want to run this:
+
+```shell
+$ cargo +nightly install cargo-fix --git https://github.com/killercup/rustfix.git
+```
+
+And that's it! Note that you do need a nightly Cargo to use `rustfix` at the
+moment; in fact, all of the edition stuff is nightly only for now. Eventually
+this will be stable! Same with installing from `git`, for now, it's needed.
+
+## Prepare for the next edition
+
+There are some lints that can help you prepare for the next edition, but
```

Comment:
It may be worth re-emphasizing a few points here as well:

* The 2018 edition has some features that are exclusive to it, aka not available in the 2015 edition. For example async/await.
* It's our intention that the migration to the 2018 is as smooth an experience as possible. It's a technical breaking change so it must be opt-in, but we want to make it as easy as possible for you to opt in so you can get access to the new features. (hence `cargo fix`)
* It's considered a bug if it's difficult to upgrade your crate to the 2018 edition. If you have a difficult time then a bug should be filed with either rustfix or rust-lang/rust

Reply:
Yes, I mentioned this in gitter. We should ask people to use `#![feature(crate_in_paths)]` before they trigger rustfix

 We can also change https://github.com/rust-lang/rust/blob/master/src/librustc_resolve/lib.rs#L3354-L3355 to no longer require crate_in_paths, but they need to add it anyway so it's not that useful? I'm less fond of the idea of lints that suggest broken code (which is what it is until crate_in_paths stabilizes)

Reply:
After some [gitter discussion](https://gitter.im/rust-lang/WG-epoch?at=5af45ceff04ce53632c8d65d) I've added a new step to the [cheat sheet](https://paper.dropbox.com/doc/Edition-preview-cheat-sheet-HhfGRmTX4mByEOWWI2P7o) where for the *preview only* users will be required to list `#![feature(rust_2018_preview)]` in their crates they wish to migrate.